river/linear_model/glm.py

```diff
@@ -74,8 +80,65 @@ def _fit(self, x, y, w, get_grad):
         # Update the weights
         self.optimizer.step(w=self._weights, g=gradient)
 
+        if (self.l1 != 0.0) and (self.l2 == 0.0):
+            # REFLECTION: this should be stated after the learning_rate update in case of adaptive learning_rate
+            self.max_cum_l1 = self.max_cum_l1 + self.l1 * self.optimizer.learning_rate
+            # /
+
+            self._update_weights(x)
+            # pass
+
         return self
 
+    def _update_weights(self, x):
+        # INFO: L1 penalty helper
+
+        for j, xj in x.items():
+            # REFLECTION: it's not `temperature` it's `temporary` lol
+            wj_temp = self._weights[j]
+            self._apply_penalty(j, wj_temp)
+
+        # INFO: perhaps a viable vectorized alternative
+        # self._apply_penalty_vecrotized(x, self._weights)
+
+    def _apply_penalty(self, j, wj_temp):
```

Comment:
I think the `def _apply_penalty()` should be defined within the `_update_weights()`, since this is the only function that uses `_apply_penalty()`.